Source/FoundationExtensions.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  FoundationExtensions.swift
+//  Rex
+//
+//  Created by Ilya Laryionau on 10/05/15.
+//  Copyright (c) 2015 Neil Pankey. All rights reserved.
+//
+
+import ReactiveCococa
+
+extension NSData {
+    /// Read the data at the URL.
+    /// Sends the data or the error.
+    class func rac_dataWithContentsOfURL(url: NSURL, options: NSDataReadingOptions = NSDataReadingOptions.allZeros) -> SignalProducer<NSData, NSError> {
+        return SignalProducer<NSData, NSError> { observer, disposable in
+            var error: NSError?
+            if let data = NSData(contentsOfURL: url, options: options, error: &error) {
+                sendNext(observer, data)
+                sendCompleted(observer)
+            } else {
+                sendError(observer, error!)
```

Comment:
Instead of force unwrapping the error can you do `error ?? NSError()`. The way [Cocoa handles errors is lame and this can be nil](https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/ErrorHandlingCocoa/CreateCustomizeNSError/CreateCustomizeNSError.html).

> Important: Success or failure is indicated by the return value of the method. Although Cocoa methods that indirectly return error objects in the Cocoa error domain are guaranteed to return such objects if the method indicates failure by directly returning nil or NO, you should always check that the return value is nil or NO before attempting to do anything with the NSError object.